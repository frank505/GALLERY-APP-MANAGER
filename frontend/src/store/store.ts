import { createStore, applyMiddleware, StoreEnhancerStoreCreator, Store } from 'redux';
import thunk from 'redux-thunk';
import {RootReducer} from './Reducers/RootReducer';
 import { HttpInterceptor } from '../apicalls/HttpInterceptor';

export const middlewares:Array<any> = [thunk];

export const createStoreWithMiddleware:StoreEnhancerStoreCreator<any> = applyMiddleware(...middlewares)(createStore)

export const store:Store = createStoreWithMiddleware(RootReducer)

 HttpInterceptor(store); 



  


