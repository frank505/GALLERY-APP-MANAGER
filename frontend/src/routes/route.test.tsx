import React from "react";
import { render, fireEvent, waitForElement, waitFor } from "@testing-library/react";
import Routes from './route';
import { Provider } from "react-redux";
import {store} from '../store/store'
import { HashRouter,  Route, Switch, Redirect } from 'react-router-dom';
import {MemoryRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';




const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
   ...jest.requireActual('react-router-dom') as any,
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

const renderRoutesComponent = () =>
{
  const Comp =  render(
    <Provider store={store}>
         <MemoryRouter initialEntries={['/test']} initialIndex={0}>
         <Routes />
         </MemoryRouter>
    </Provider>    
    );

    return Comp;
}






describe('it should mock route', () => {
    
    test('renders Route component as expected',()=>{
      renderRoutesComponent();
    });

    test('renders component correctly',async()=>
    {
        const tree = renderer
        .create(<Routes/>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    })

})
