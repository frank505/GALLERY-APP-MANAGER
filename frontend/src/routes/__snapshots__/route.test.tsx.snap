// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should mock route renders component correctly 1`] = `
<div>
  this is the login component
</div>
`;
